diff --git a/src/main/scala/v3/exu/execution-units/functional-unit.scala b/src/main/scala/v3/exu/execution-units/functional-unit.scala
index 92fe8219..b12a4303 100644
--- a/src/main/scala/v3/exu/execution-units/functional-unit.scala
+++ b/src/main/scala/v3/exu/execution-units/functional-unit.scala
@@ -449,10 +449,10 @@ class ALUUnit(isJmpUnit: Boolean = false, numStages: Int = 1, dataWidth: Int)(im
   val r_data = RegInit(VecInit(Seq.fill(numStages)(0.U(xLen.W))))
   val r_pred = RegInit(VecInit(Seq.fill(numStages)(false.B)))
   val alu_out = //Mux(io.req.bits.uop.ctrl.op_fcn === 0.U, Cat(alu.io.out(31,1), 0.U(1.W)),
-    //Mux(io.req.bits.uop.ctrl.op_fcn === 10.U, Cat( 0.U(1.W), alu.io.out(62,0)),
+    Mux(io.req.bits.uop.ctrl.op_fcn === 10.U, Cat( 0.U(1.W), alu.io.out(62,0)),
     Mux(io.req.bits.uop.is_sfb_shadow && io.req.bits.pred_data,
     Mux(io.req.bits.uop.ldst_is_rs1, io.req.bits.rs1_data, io.req.bits.rs2_data),
-    Mux(io.req.bits.uop.uopc === uopMOV, io.req.bits.rs2_data, alu.io.out))
+    Mux(io.req.bits.uop.uopc === uopMOV, io.req.bits.rs2_data, alu.io.out)))
   r_val (0) := io.req.valid
   r_data(0) := Mux(io.req.bits.uop.is_sfb_br, pc_sel === PC_BRJMP, alu_out)
   r_pred(0) := io.req.bits.uop.is_sfb_shadow && io.req.bits.pred_data
